File: app/src/main/java/com/xingen/camera/base/BaseApplication.java
package com.xingen.camera.base;

import android.app.Application;

/**
 * Created by ${xingen} on 2017/10/20.
 */

public class BaseApplication extends Application {
    private static BaseApplication instance;
    @Override
    public void onCreate() {
        super.onCreate();
        instance=this;
    }
    public static BaseApplication getInstance() {
        return instance;
    }
}

File: app/src/main/java/com/xingen/camera/base/BaseApplication.kt
package com.xingen.camera.base

import android.app.Application

/**
 * Created by ${xingen} on 2017/10/20.
 */
class BaseApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    companion object {
        @JvmStatic
        var instance: BaseApplication? = null
            private set
    }
}
